### DISCUSSION TITLE
Consider using `pybind11` in C-backend

### DISCUSSION THREAD

**Discussion by ferrine** [aesara-devs/aesara#1052] [Ideas]
The idea is to simplify our c++ backend using
https://github.com/tbenthompson/cppimport
and 
https://github.com/pybind/pybind11

**Comment by brandonwillard** [comment]
See https://github.com/aesara-devs/aesara/discussions/376#discussioncomment-621963 first.  If we could use Cython for the C-backend, we would solve numerous issues and make it very easy to add new and improved C implementations.  That's currently the highest priority for C-backend refactoring.
